### ISSUE TITLE
Add a way to filter out tests in the Editor window - by namespace or category

### ISSUE THREAD

**Issue by rucindrea** [alttester/AltTester-Unity-SDK#1330]
## Description
When a project contains already a large number of Unity Test Framework tests, the "AltTesterEditorSettings.asset" can get quite big and the AltTester editor window shows a lot of tests that are not AltTester tests. 

We should hav ea way to filter tests by category or by namespace, so that the "AltTesterEditorSettings.asset" would also only include info about those tests.

**Comment by iulianatodoran** [comment]
### Testing environment
- _SampleGame_ project with AltTester Unity SDK from `development` branch (27.07.2023)
- Unity v `2020.3.48f1` LTS, `2021.3.25f1` LTS, `2022.3.5f1` LTS

### Tests performed and results
Checked in the AltTester Editor window that:
- filter is applied correctly while playing/not playing in Editor with `ALTTESTER` symbol checked/unchecked in Unity 20
  - ❌ **BUT** **AR**: in Unity 2020 and 2022 the test `DefaultTestExample.txt` is not displayed at all in its correspondent  `AltTester.AltTesterUnitySDK.Editor` assembly 
  - ❌ **BUT** **AR**: in Unity 2021 the test `DefaultTestExample.txt` is misplaced in the `AltTester.AltTesterUnitySDK` assembly > **ER**: it should be placed in the `AltTester.AltTesterUnitySDK.Editor` assembly
- ✅ Tests list can be refreshed or the filter applied while playing in Editor with `ALTTESTER` symbol checked/unchecked
- ✅ filter contains all assemblies that have NUnit reference in Unity in Unity 2021 LTS
  - ❌ **BUT** **AR**: there is one assembly that is not displayed in Unity 2020 and 2022 LTS:  `AltTester.AltTesterUnitySDK`
- ✅ all tests are displayed for the chosen assembly
- ✅ all tests from all assemblies are displayed when the `All` filter is applied
- ✅ the tests can be run as expected from the `Tests list`
- ✅ the Tests list refreshes when clicking on the `Refresh` button
- ✅ when a new test file is created it is displayed in the corresponding assembly category after refreshing the `Tests list`
- ✅ all the functionalities work as expected and no issues are brought from the new implementation in the `Tests list` section
- ✅ the filter works as expected in different LTS Unity versions supported with AltTester (2020, 2021, 2022)
- ✅ there are no errors related to the AltTester Editor window and AltTester Unity SDK in Unity 2021 and 2022 LTS
  - ❌ **BUT** **AR**: there is a `C#` syntax error in Unity 2020: `Assets/AltTester/Runtime/Input.cs(1334,24): error CS0246: The type or namespace name ‘not’ could not be found (are you missing a using directive or an assembly reference?)` 
      ![image](https://github.com/alttester/AltTester-Unity-SDK/assets/62327834/c029f700-27ef-4925-a910-db011a35364c)
- ✅ there are no display/GUI issues
- ✅ there are no theme-related issues in Unity 2020 LTS
  - ❌ **BUT** **AR**: the filter menu uses a white theme color for the dark theme in Unity 2021 and 2022 LTS
      ![image](https://github.com/alttester/AltTester-Unity-SDK/assets/62327834/684d4bfc-f6d9-4b15-bc32-bd367e48aba2)

### Notes on testing
- encountered also Unity-related errors when trying to open test files that were not included under an `Editor` folder and only after that other Unity errors if trying to use the tests filter for assemblies that do not contain tests , but this is not related to AltTester

### Conclusion
For the above-mentioned issue will re-open the ticket.

**Comment by RobertPoienar** [comment]
I will not fixed the following: theme related issue because is an Unity bug and the assembly not appearing/test misplaced. This are minor issues does are not something we can control and it's not worth the time for this release.

@iulianatodoran could you please test if you have a real test in the `AltTester.AltTesterUnitySDK` does the assembly appear in the list.

**Comment by iulianatodoran** [comment]
### Re-testing environment
- _SampleGame_ project with AltTester Unity SDK from `development` branch (01.08.2023)
- Unity v `2020.3.48f1` LTS, `2021.3.25f1` LTS, `2022.3.5f1` LTS

### Tests performed and results
Checked in the AltTester Editor window that:
- ✅ for creating a test file inside the `AltTester.AltTesterUnitySDK` assembly it will appear in the `Tests list` inside the correspondent assembly category (in all unity versions)
- ✅ there are no errors related to the AltTester Editor window and AltTester Unity SDK in all Unity versions
- ✅ all the main functionalities work as expected
- ✅ there are no other issues related to the filter than those that were not able to be fixed for this release (as mentioned above)

### Notes
- the issues mentioned in the first comment that could not be fixed for this release will be the subject of another ticket [to be mentioned here after opening)

### Conclusions
The purpose of the ticket was achieved and all other issues found will be fixed in another ticket as agreed.

**Closing the ticket.**
